src/mongo/tree/MongoDocumentTreeItem.ts

```diff
@@ -58,8 +58,13 @@ export class MongoDocumentTreeItem implements IAzureTreeItem {
     }
 
     public async update(newDocument: IMongoDocument): Promise<IMongoDocument> {
-        this.document = await MongoDocumentTreeItem.update(this._collection, newDocument);
-        return this.document;
+        if (Object.getOwnPropertyNames(newDocument).length === 0) {
+            throw new Error("Updating a document with no fields in the document does nothing");
```

Comment:
Same comment as above (Except this time check for _id instead of _self/_etag).

Also move the check down into the static update method instead of here (otherwise anything that calls the static method directly won't get this message).